src/Host/Monitor/Impl/BrokerManager.cs

```diff
@@ -10,16 +11,16 @@ namespace Microsoft.R.Host.Monitor {
     public class BrokerManager {
```

Comment:
Actually, so far as I can see, the only two places where UI is updated here is at the end of `CreateOrAttachToBrokerInstanceAsync` and `StopBrokerInstanceAsync`. So instead of doing the update here, just change it so that the caller does it once the task completes successfully.